analyzers/src/SonarAnalyzer.CSharp/Rules/FrameworkViewCompiler.cs

```diff
@@ -0,0 +1,339 @@
+        }
+            .SelectMany(x => x.Assembly.GetExportedTypes())
+            .ToArray();
+        CodeFixTypes = allTypes.Where(x => typeof(CodeFixProvider).IsAssignableFrom(x) && x.GetCustomAttributes<ExportCodeFixProviderAttribute>().Any()).ToArray();
+        AllAnalyzerTypes = allTypes.Where(x => x.IsSubclassOf(typeof(DiagnosticAnalyzer)) && x.GetCustomAttributes<DiagnosticAnalyzerAttribute>().Any()).ToArray();
+        AllTypesWithDiagnosticAnalyzerAttribute = allTypes.Where(x => x.GetCustomAttributes<DiagnosticAnalyzerAttribute>().Any()).ToArray();
+        UtilityAnalyzerTypes = AllAnalyzerTypes.Where(x => typeof(UtilityAnalyzerBase).IsAssignableFrom(x)).ToList();
+        RuleAnalyzerTypes = AllAnalyzerTypes.Except(UtilityAnalyzerTypes).ToList();
+    }
+
+    public static IEnumerable<Type> GetAnalyzerTypes(AnalyzerLanguage language) =>
+        RuleAnalyzerTypes.Where(x => TargetLanguage(x) == language);
+
+    public static IEnumerable<DiagnosticAnalyzer> CreateAnalyzers(AnalyzerLanguage language, bool includeUtilityAnalyzers)
+    {
+        var types = GetAnalyzerTypes(language);
+        if (includeUtilityAnalyzers)
+        {
+            types = types.Concat(UtilityAnalyzerTypes.Where(x => TargetLanguage(x) == language));
+        }
+        foreach (var type in types.Where(x => x == typeof(VariableUnused))) // TODO: Remove ".Where(x => x == typeof(VariableUnused))" and see error. Locally it breaks for me with StackOverflow, at a compiled Regex.
+        {
+            yield return typeof(HotspotDiagnosticAnalyzer).IsAssignableFrom(type) && type.GetConstructor(new[] { typeof(IAnalyzerConfiguration) }) != null
+                ? (DiagnosticAnalyzer)Activator.CreateInstance(type, AnalyzerConfiguration.AlwaysEnabled)
+                : (DiagnosticAnalyzer)Activator.CreateInstance(type);
+        }
+    }
+
+    public static bool IsParameterized(Type analyzerType) =>
+        analyzerType.GetProperties().Any(x => x.GetCustomAttributes<RuleParameterAttribute>().Any());
+
+    private static AnalyzerLanguage TargetLanguage(MemberInfo analyzerType)
+    {
+        var languages = analyzerType.GetCustomAttributes<DiagnosticAnalyzerAttribute>().SingleOrDefault()?.Languages
+            ?? throw new NotSupportedException($"Can not find any language for the given type {analyzerType.Name}!");
+        return languages.Length == 1
+            ? AnalyzerLanguage.FromName(languages.Single())
+            : throw new NotSupportedException($"Analyzer can not have multiple languages: {analyzerType.Name}");
+    }
+}
+
+#endregion
+
+#region RAZOR_COMPILER
+
+// Copied from sonar-security.
+
+/// <summary>
+/// Compiler for manual Razor Views cross-compilation to C# code for .NET Framework projects.
+/// We don't need to manually compile .NET Core Razor projects. They are compiled during build time and analyzed directly with Roslyn analyzers.
+/// </summary>
+internal sealed class RazorCompiler
```

Comment:
This is the magic/hack part.
This package cross-compiles cshtml -> c#, but only for .NET Core.

So we do a lot of dirty things to make it play a bit better with the .NET FM type system.